script/tool/lib/src/update_excerpts_command.dart

```diff
@@ -23,232 +26,302 @@ class UpdateExcerptsCommand extends PackageLoopingCommand {
-        'dart',
-        <String>[
-          'run',
-          'code_excerpt_updater',
-          '--write-in-place',
-          '--yaml',
-          '--no-escape-ng-interpolation',
-          ...relativeMdPaths,
-        ],
-        workingDir: example.directory);
-    return exitCode == 0;
-  }
 
-  /// Adds `code_excerpter` and `code_excerpt_updater` to [package]'s
-  /// `dev_dependencies` using path-based references to the submodule copies.
-  ///
-  /// This is done on the fly rather than being checked in so that:
-  /// - Just building examples don't require everyone to check out submodules.
-  /// - Examples can be analyzed/built even on versions of Flutter that these
-  ///   submodules do not support.
-  void _addSubmoduleDependencies(RepositoryPackage package,
-      {required Directory repoRoot}) {
-    final String pubspecContents = package.pubspecFile.readAsStringSync();
-    // Save aside a copy of the current pubspec state. This allows restoration
-    // to the previous state regardless of its git status at the time the script
-    // ran.
-    package.directory
-        .childFile(_originalPubspecFilename)
-        .writeAsStringSync(pubspecContents);
-
-    // Update the actual pubspec.
-    final YamlEditor editablePubspec = YamlEditor(pubspecContents);
-    const String devDependenciesKey = 'dev_dependencies';
-    final YamlNode root = editablePubspec.parseAt(<String>[]);
-    // Ensure that there's a `dev_dependencies` entry to update.
-    if ((root as YamlMap)[devDependenciesKey] == null) {
-      editablePubspec.update(<String>['dev_dependencies'], YamlMap());
+    String pathBase = path.dirname(file.path);
+    final StringBuffer output = StringBuffer();
+    final StringBuffer existingBlock = StringBuffer();
+    String? language;
+    String? excerpt;
+    _ExcerptParseMode mode = _ExcerptParseMode.normal;
+    int lineNumber = 0;
+    for (final String line in file.readAsLinesSync()) {
+      lineNumber += 1;
+      switch (mode) {
+        case _ExcerptParseMode.normal:
+          if (line.contains(_pragma)) {
+            RegExpMatch? match = _basePattern.firstMatch(line);
+            if (match != null) {
+              pathBase = match.group(1)!;
+            } else {
+              match = _injectPattern.firstMatch(line);
+              if (match != null) {
+                final String excerptPath = match.namedGroup('path')!;
+                final File excerptSourceFile = file.fileSystem.file(
+                    path.normalize(path.join(
+                        path.dirname(file.path), pathBase, excerptPath)));
+                final String extension = path.extension(excerptSourceFile.path);
+                switch (extension) {
+                  case '':
+                    language = 'txt';
+                    break;
+                  case '.kt':
+                    language = 'kotlin';
+                    break;
+                  case '.cpp':
```

Comment:
We also have `.cc` files; the Linux and Windows conventions are, unfortunately, different.

Reply:
added